src/OrchardCore.Modules/Orchard.Liquid/Startup.cs

```diff
@@ -0,0 +1,89 @@
+using System.Linq;
+using System.Net;
+using System.Security.Principal;
+using DotLiquid;
+using DotLiquid.NamingConventions;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Internal;
+using Microsoft.AspNetCore.Modules;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json.Linq;
+using Orchard.ContentManagement;
+using Orchard.ContentManagement.Display.ContentDisplay;
+using Orchard.ContentManagement.Handlers;
+using Orchard.Data.Migration;
+using Orchard.Indexing;
+using Orchard.Liquid.Drivers;
+using Orchard.Liquid.Drops;
+using Orchard.Liquid.Filters;
+using Orchard.Liquid.Handlers;
+using Orchard.Liquid.Indexing;
+using Orchard.Liquid.Model;
+
+namespace Orchard.Liquid
+{
+    public class Startup : StartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddLiquid();
+
+            // Liquid Part
+            services.AddScoped<IContentPartDisplayDriver, LiquidPartDisplay>();
+            services.AddSingleton<ContentPart, LiquidPart>();
+            services.AddScoped<IDataMigration, Migrations>();
+            services.AddScoped<IContentPartIndexHandler, LiquidPartIndexHandler>();
+            services.AddScoped<IContentPartHandler, LiquidPartHandler>();
+
+            Template.RegisterSafeType(typeof(IContent), new string[] { nameof(IContent.ContentItem) });
+            Template.RegisterSafeType(typeof(ContentItem), o => new ContentItemDrop((ContentItem)o));
+
+            Template.RegisterSafeType(typeof(JToken), o => new JTokenDrop((JToken)o));
+            Template.RegisterSafeType(typeof(JObject), o => new JTokenDrop((JToken)o));
```

Comment:
Maybe here we would need this

                Template.RegisterSafeType(typeof(JObject), o => new JTokenDrop((JObject)o));

I first thought that doing this will solve the issue related to your comment in `QueriesDrop.cs`

> // BUG: Somehow Liquid doesn't accept JObject collections directly unlike ContentItem.

But i may be wrong or it is not sufficient. Oh, maybe we need to create a `JObjectDrop`, and then use.

                Template.RegisterSafeType(typeof(JObject), o => new JObjectDrop((JObject)o));